pkg/scheduler/core/generic_scheduler.go

```diff
@@ -1162,7 +1163,11 @@ func nodesWherePreemptionMightHelp(nodes []*v1.Node, failedPredicatesMap FailedP
 // considered for preemption.
 // We look at the node that is nominated for this pod and as long as there are
 // terminating pods on the node, we don't consider this for preempting more pods.
-func podEligibleToPreemptOthers(pod *v1.Pod, nodeNameToInfo map[string]*schedulernodeinfo.NodeInfo) bool {
+func podEligibleToPreemptOthers(pod *v1.Pod, nodeNameToInfo map[string]*schedulernodeinfo.NodeInfo, enableNonPreempting bool) bool {
+	if enableNonPreempting && pod.Spec.Preempting != nil && !*(pod.Spec.Preempting) {
```

Comment:
if we encounter a non-nil preemptionPolicy we don't understand here, I would recommend not preempting (and logging if we can do so without flooding the logs)

Reply:
Done